js-pkg/apps/demos/src/components/webrtc.js

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react'
+import { useDriftDBSignalingChannel } from './driftdbutils'
+
+export const useWebRTCMessagingChannel = (p1, p2) => {
+  let [setConnSetupArray] = useWebRTCConnection(p1, p2)
+  let [theyAreReady, setTheyAreReady] = React.useState(false)
+  let [messages, addMessage] = React.useReducer((state, msg) => [...state, msg], [])
+  let theirdataChannelRef = React.useRef(null)
+
+  React.useEffect(() => {
+    setConnSetupArray([
+      (conn) => {
+        let dataChannel = conn.createDataChannel(p1)
+        dataChannel.onmessage = (e) => {
+          addMessage({ id: e.timeStamp, text: e.data })
+        }
+
+        conn.ondatachannel = (e) => {
+          if (e.channel.label === p2) {
+            setTheyAreReady(true)
+            theirdataChannelRef.current = e.channel
+          }
+        }
+      }
+    ])
+
+    return () => {
+      setTheyAreReady(false)
+    }
+  }, [p1, p2])
+
+  return [
+    messages,
+    theyAreReady
+      ? theirdataChannelRef.current.send.bind(theirdataChannelRef.current)
+      : (msg) => {
+          console.log('unsent: ', msg)
+        }
```

Comment:
from offline convo: you could get rid of the `theyAreReady` state above and just rely on the `theirDataChannelRef`

```js
(msg) => {
  if (theirdataChannelRef.current) theirdataChannelRef.current.send(msg)
  else console.log('unsent:', msg)
}
```